packages/react/src/hooks/useInfiniteScroll.ts

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useRef, useState } from 'react';
+import useWindowSize from './useWindowSize';
+
+export interface Queries {
+  page: number;
+  pageSize: number;
+}
+export interface InfiniteScroll<T> {
+  fetcher: (params: Queries) => Promise<T>;
+  initialQueries: Queries;
+  getQueries: (params?: any) => Queries;
+  threshold?: number;
+}
+
+/**
+ *  useInfiniteScroll hook
+ *
+ *  threshold 에 닿았을때 fetch 함수를 실행합니다.
+ *
+ * @param {boolean} isLoading 로딩 상태
+ * @param {boolean} hasNextPage fetch 데이터의 유무
+ * @param {Function} onLoadMore threshold 에 닿았을때 호출 할 함수
+ * @param {number} threshold onLoadMore 이 호출 될 높이
+ * @param {number} interval 업데이트 간격
+ */
+
+const useInfiniteScroll = <T>({
+  fetcher,
+  initialQueries,
+  getQueries,
+  threshold = 150,
+}: InfiniteScroll<T>) => {
+  const ref = useRef<HTMLElement>(null);
+  const { height: windowHeight } = useWindowSize();
+  const [isLoading, setIsLoading] = useState(false);
+  const [data, setData] = useState<T>();
+  const [query, setQuery] = useState(initialQueries);
+
+  const fetchData = () => {
+    const nextQuery = getQueries(query);
+    setQuery(nextQuery);
+
+    return fetcher(nextQuery);
+  };
+
+  const getBottomOffset = () => {
+    const element = ref.current;
+
+    if (!element || !windowHeight) {
+      return null;
+    }
+
+    const rect = element.getBoundingClientRect();
+
+    const bottom = rect.bottom;
+    const bottomOffset = bottom - windowHeight;
+
+    return bottomOffset;
+  };
+
+  const listenBottomOffset = async () => {
+    if (!isLoading) {
```

Comment:
```suggestion
    if (isLoading) {
      return;
    }
```

요렇게 Ealry Return 패턴으로 작성하면 `if` 문 중첩을 없앨수도 있고, 코드를 읽을 때도 예외조건을 하나씩 지워나가며 생각하기 편할 것 같습니다!

Reply:
증말 무지성으로 코딩했네용...

https://github.com/Lubycon/lubycon-frontend-libraries/pull/45/commits/f2bf938ac1f844eec7fe197482d4364158c98b42 반영했슴당